### ISSUE TITLE
macOS Sierra: Invalid Provisioning Profile

### ISSUE THREAD

**Issue by bradleyduncan** [DanTheMan827/ios-app-signer#19]
![screen shot 2016-06-14 at 2 59 13 pm](https://cloud.githubusercontent.com/assets/6460526/16047381/bb83a3de-3247-11e6-8d23-c8c7eddedb04.png)
Since updating to macOS Sierra (Build) 16A201, I'm unable to sign applications using a provisioning profile, although before downloading the beta, this was not an issue. 

Thank you so much for all your hard work.

**Comment by DanTheMan827** [comment]
Are you also using Xcode 8.0 beta?

**Comment by bradleyduncan** [comment]
I have both installed and have compiled the app in both.

**Comment by bradleyduncan** [comment]
Reproduction
- Open iOS App Signer
- Select 'Choose Custom File' from the 'Provisioning Profile' menu
- Error produced: Invalid provisioning profile 

This is before selecting the 'Input File'

**Comment by DanTheMan827** [comment]
That sounds like the open file dialog isn't working...

As a workaround you should be able to put your profile in ~/Library/MobileDevice/Provisioning Profiles

That way it should get picked up by iOS App Signer without needing to choose the file manually

Please verify if that works or not.

I'll have to install Mac OS Sierra DP1 before I can even attempt to fix it though...

**Comment by bradleyduncan** [comment]
iOS App Signer didn't detect the Profile when but in the folder ~/Library/MobileDevice/Provisioning Profiles. Only these options appear: 
![screen shot 2016-06-14 at 4 21 25 pm](https://cloud.githubusercontent.com/assets/6460526/16048466/1a6043cc-324c-11e6-9bde-0806f6ae7983.png)

Here is the provisioning file in the directory you requested:
![screen shot 2016-06-14 at 4 22 32 pm](https://cloud.githubusercontent.com/assets/6460526/16048503/3ff49f52-324c-11e6-8a16-cd70a95487f2.png)

The workaround didn't work. Let me know if there is any more tests you wish me to do in the meanwhile 🙂

**Comment by DanTheMan827** [comment]
Could you post the iOS App Signer Log? (View > Application Log)

**Comment by bradleyduncan** [comment]
## iOS App Signer Log
2016-06-14 16:37:44 Error parsing 31e7127e-c760-4902-95fe-979780bc5130.mobileprovision
2016-06-14 16:37:44 Error parsing d297e8fc-c0a5-4891-86e2-7df78b93b4e0.mobileprovision
2016-06-14 16:37:44 Error parsing Profile.mobileprovision
2016-06-14 16:37:44 Found 0 Provisioning Profiles
2016-06-14 16:37:45 Found 5 Codesigning Certificates
2016-06-14 16:37:45 Added signing certificate "Mac Developer: example1@gmail.com (YEQ66YY8GK)"
2016-06-14 16:37:45 Added signing certificate "iPhone Developer: example2@live.co.uk (2Q5SSKDE3E)"
2016-06-14 16:37:45 Added signing certificate "iPhone Developer: example3@icloud.com (A2KNPYKV45)"
2016-06-14 16:37:45 Added signing certificate "iPhone Developer: example1@gmail.com (YEQ66YY8GK)"
2016-06-14 16:37:45 Added signing certificate "iPhone Developer: Kevin Swanson (R2M22ZA75C)"
2016-06-14 16:37:45 Loaded Codesigning Certificate from Defaults: iPhone Developer: Kevin Swanson (R2M22ZA75C)
2016-06-14 16:37:45 Ready
2016-06-14 16:37:56 Error parsing Profile.mobileprovision
2016-06-14 16:37:56 Invalid provisioning profile

**Comment by DanTheMan827** [comment]
Could you try running this command and see if it reports any error

Adjust the path to Profile.mobileprovision as needed

/usr/bin/security cms -D -i ~/Desktop/Profile.mobileprovision

**Comment by bradleyduncan** [comment]
##Result
`security: SecPolicySetValue: One or more parameters passed to a function were not valid.`

It then proceeds to list the certificate information.

**Comment by DanTheMan827** [comment]
okay... so they changed the usage of security...

I'll have to do some investigating...

**Comment by bradleyduncan** [comment]
Is there any other information I could provide in the meanwhile?

**Comment by bradleyduncan** [comment]
Read this article: http://arstechnica.co.uk/apple/2016/06/apfs-apple-new-file-system-dev-details/

It may be relevant to building a solution; or may not! I'm not sure! 😀

**Comment by navysandsquid** [comment]
is there any update on this?  Could use a fix

**Comment by DanTheMan827** [comment]
I haven't had the time... I don't even have MacOS Sierra installed yet.. :/

**Comment by navysandsquid** [comment]
ok,  I'm in a hotel and trying to get a web browser on my apple tv 4 if you had any ideas.  the torrents is pretty good for macOS

**Comment by zanesavage** [comment]
First off, great app! Thanks.

I'm having the same issue; any fix yet?

Cheers

**Comment by NoahSaso** [comment]
Hey guys! I just wanted to let you all know I created a temporary fix for this bug on macOS Sierra until DanTheMan827 can adjust the actual command to work. Here's my pull request: https://github.com/DanTheMan827/ios-app-signer/pull/20
You should be able to clone that and build it in Xcode without any hassle. Lemme know how it goes!

**Comment by bradleyduncan** [comment]
Thanks so much NoahSaso and DanTheMan for your work! Confirmed the fix to be working! If you follow the link Noah posted, it has the edits to the Xcode project. Anyone else who's feeling lazy, here's the pre-built app link https://mega.nz/#!4oAjzQjA

**Comment by navysandsquid** [comment]
awesome looking forward to trying it whats the passcode to the mega link?

**Comment by obi1andkobe** [comment]
Thanks bradleyduncan. Can you share the passcode/decryption key for the mega upload file?

**Comment by zanesavage** [comment]
Thanks bradleyduncan. Can you send me the passcode.

**Comment by NoahSaso** [comment]
@navysandsquid @obi1andkobe @zanesavage If you guys didn't get the password or for anyone else, I put the compiled app up on my server. Should've done that sooner haha
Here's the link: [http://files.noahsaso.com/iOS%20App%20Signer.zip](http://files.noahsaso.com/iOS%20App%20Signer.zip)

**Comment by bradleyduncan** [comment]
https://mega.nz/#!4oAjzQjA!est5RfNBhzJlMh1kXADbfC6_BxZJbEDSyj1kgoaC8Dk Apologies on encryption issue!

**Comment by navysandsquid** [comment]
THANKS!

**Comment by Bigperm1** [comment]
Thanks works Great

**Comment by machmudi** [comment]
It works with previous Command Line Tools
Here's a link https://mega.nz/#!fBwQ0bjJ!-1qerSM85RxsFRDURL5iP7MR37GiA6aSeOWDzEk8qIo

**Comment by dmf6978** [comment]
hi guys. new to the whole iOS app signer thing. last week installed kodi beta and now i have to go back to version 16. i am running sierra and 8.1 Xcode i do the steps properly. and then iOS app signer takes about 3 min to open the program. i select the items i need to and then it just seems to freeze and not export the file i need to then upload to my apple tv. any ideas why? i have erased the program and preferences many times.

**Comment by wessidetempest** [comment]
So I am running into a similar issue - slightly different. iOS is not showing the option of the Provisioning Profile. the iOS log shows that it is there and correct.  In the dropdown try selecting Custom Profile I can never navigate to ~/Library/MobileDevice/Provisioning Profiles.  I went so far to copy it to desktop and it did not work. I downloaded the link above but not sure what to do with it. And in this case it is the latest Xcode and Mac OS.  I am desperate simply because I want to figure it out and I can't
